reference/language-concepts/namespaces.markdown

```diff
@@ -6,104 +6,76 @@ sorting: 100
+places the bundle or body in a different namespace to allow re-use of common
+names. Using namespaces makes it easier to share and consume policy from other
+authors.
 
-Everything in CFEngine lives in a namespace (it's the `default` namespace if not set).
+Like bundle names and classes, namespaces may only contain alphanumeric and
+underscore characters (`a-zA-Z0-9_`).
 
-### Specifying a namespace
+## Declaration
 
-To isolate a file into its own namespace, you add a
-[file control promise][file control#namespace] to the file before the relevant
-bundles or bodies. All bundles and bodies start off in the `default` namespace
-if you don't explicitly set this. Once set, this applies until the end of the
-file or the next namespace change.
+Namespaces are declared with [`body file control`][file control#namespace]. A
+namespace applies within a single file to all subsequently defined bodies
+following the namespace declaration until a different namespace has been
+declared or until the end of the file.
 
-```cf3
-    body file control
-    {
-       namespace => "myspace";
-    }
-```
+[%CFEngine_include_example(namespace_declaration.cf)%]
 
-### Accessing syntax elements between namespaces and the default namespace
+**Notes:**
 
-To distinguish the bundle `mymethod` in the default namespace from one in
-another namespace, you prefix the bundle name with the namespace, separated by
-a colon.
+- Multiple namespaces can be declared within the same file
+- The same namespace can be declared in multiple files
+- The same namespace can be declared in the same file multiple times
 
-```cf3
-    methods:
+## Methods|usebundle
 
-      "namespace demo" usebundle => myspace:mymethod("arg1");
-      "namespace demo" usebundle => mymethod("arg1","arg2");
-```
+Methods promises assume you are referring to a bundle in the same namespace as
+the promiser. To refer to a bundle in another namespace you *must* specify the
+namespace by prefixing the bundle name with the namespace followed by a colon
+(`:`).
 
-To distinguish a body from one in another namespace, you can prefix the body name with the namespace, separated by a colon.
+[%CFEngine_include_example(namespace_methods-usebundle.cf)%]
 
-```cf3
-    files:
-       "/file"
-          create => "true",
-           perms => name1:settings;
-```
+## Bodies
 
-If you don't make any namespace declarations, you'll be in the
-`default` namespace.  Bundles, bodies, classes, and variables from the
-`default` namespace can be accessed like any other:
+Bodies are assumed to be within the same namespace as the promiser. To use a body from another namespace the namespace must be specified by prefixing the body name with the namespace followed by a colon (`:`).
 
-```cf3
-    files:
-      "/file"
-         create => "true",
-          perms => default:settings;
-```
+A common mistake is forgetting to specify `default:` when using bodies from the standard library which resides in the `default` namespace.
```

Comment:
This kind of mentions what I was saying earlier but I think you could more clearly link the inclusion of a namespace in `body file control` to everything after needing `default:` to use standard library items.

Reply:
Yeah, also embedded in some examples: https://github.com/cfengine/core/pull/4966

if you have a differnnt concrete suggestion happy to amend the text